### PR TITLE
Forms Authentication Validation - Replace IsValid with EnsureConfigurationIsValid()

### PR DESCRIPTION
**Pull request by xt0rted**
There's two breaking changes here. The first is the removal of the `IsValid` property and the second is changing the exception type in the `Enable` method.

### CODE REVIEW

**Comment by thecodejunkie** [comment]
@xt0rted so the only reason this is a breaking change is because of the `IsValid -> EnsureConfigurationIsValid()` change? That is, if we ignore the fact that someone might be catching the `ArgumentException` for `.Enable`, but you now throw `InvalidOperationException`. I think the exception is less trivial change. It's still breaking, but I would be more inclined to pull this into 1.3 if it was the only breaking change. As it stands now it would probably end up in the 2.x time frame.

To be honest I would much rather see is keep `IsValid` and be able to pull this into 1.3

File: src/Nancy.Authentication.Forms.Tests/FormsAuthenticationConfigurationFixture.cs [hunk 1]
```diff
@@ -1,9 +1,12 @@
 namespace Nancy.Authentication.Forms.Tests
 {
+    using System;
+
     using FakeItEasy;
 
     using Nancy.Cryptography;
     using Nancy.Tests;
+
```
Comment (khellang) [inline src/Nancy.Authentication.Forms.Tests/FormsAuthenticationConfigurationFixture.cs]: :fire:

File: src/Nancy.Authentication.Forms.Tests/FormsAuthenticationFixture.cs [hunk 2]
```diff
@@ -3,14 +3,16 @@ namespace Nancy.Authentication.Forms.Tests
     using System;
     using System.Linq;
     using System.Threading;
-    using Bootstrapper;
-    using Cryptography;
+
```
Comment (khellang) [inline src/Nancy.Authentication.Forms.Tests/FormsAuthenticationFixture.cs]: :fire:

File: src/Nancy.Authentication.Forms.Tests/FormsAuthenticationFixture.cs [hunk 3]
```diff
@@ -3,14 +3,16 @@ namespace Nancy.Authentication.Forms.Tests
...
     using System.Linq;
     using System.Threading;
-    using Bootstrapper;
-    using Cryptography;
+
     using FakeItEasy;
-    using Fakes;
-    using Helpers;
-    using Nancy.Security;
+
```
Comment (khellang) [inline src/Nancy.Authentication.Forms.Tests/FormsAuthenticationFixture.cs]: :fire:

File: src/Nancy.Authentication.Forms.Tests/FormsAuthenticationFixture.cs [hunk 4]
```diff
@@ -3,14 +3,16 @@ namespace Nancy.Authentication.Forms.Tests
...
+    using Nancy.Authentication.Forms.Tests.Fakes;
+    using Nancy.Bootstrapper;
+    using Nancy.Cryptography;
+    using Nancy.Helpers;
     using Nancy.Tests;
     using Nancy.Tests.Fakes;
+
```
Comment (khellang) [inline src/Nancy.Authentication.Forms.Tests/FormsAuthenticationFixture.cs]: :fire:

File: src/Nancy.Authentication.Forms/FormsAuthentication.cs [hunk 5]
```diff
@@ -1,11 +1,12 @@
 namespace Nancy.Authentication.Forms
 {
     using System;
-    using Bootstrapper;
-    using Cookies;
-    using Cryptography;
-    using Helpers;
+
```
Comment (khellang) [inline src/Nancy.Authentication.Forms/FormsAuthentication.cs]: :fire:

File: src/Nancy.Authentication.Forms/FormsAuthenticationConfiguration.cs [hunk 6]
```diff
@@ -1,6 +1,8 @@
 namespace Nancy.Authentication.Forms
 {
-    using Cryptography;
+    using System;
+
```
Comment (khellang) [inline src/Nancy.Authentication.Forms/FormsAuthenticationConfiguration.cs]: :fire:

File: src/Nancy.Authentication.Forms/FormsAuthenticationConfiguration.cs [hunk 7]
```diff
@@ -68,39 +70,35 @@ public FormsAuthenticationConfiguration(CryptographyConfiguration cryptographyCo
...
-                    return false;
-                }
-
-                return true;
+                throw new InvalidOperationException("CryptographyConfiguration HmacProvider cannot be null.");
             }
         }
     }
-}
+}
```
Comment (khellang) [inline src/Nancy.Authentication.Forms/FormsAuthenticationConfiguration.cs]: Line break, please? https://robots.thoughtbot.com/no-newline-at-end-of-file :innocent:

**Comment by khellang** [comment]
If you fix the nits and rebase this, I'll be here to pull it :wink:

**Comment by xt0rted** [comment]
@khellang  rebased and cleaned up

**Comment by khellang** [comment]
Above and beyond! :sparkles: Thanks, @xt0rted :heart:

**Pull request merged**